docs/docs/guides/events_subscriptions/custom_subscriptions.md

```diff
@@ -0,0 +1,230 @@
+
+-   `EventMap` - The event map for events emitted by the subscription
+-   `ArgsType` - The arguments passed to the subscription
+
+For example:
+
+```ts
+class MyCustomSubscription extends Web3Subscription<
+	{
+		// here provide the type of the `data` that will be emitted by the node
+		data: string,
+	},
+	// here specify the types of the arguments that will be passed to the node when subscribing
+	{
+		customArg: string,
+	},
+> {
+	// ...
+}
+```
+
+### Specify the Subscription Arguments
+
+You need to specify the exact data that will be passed to the provide. You do this by overriding `_buildSubscriptionParams` in your class. It could be something as follow:
+
+```ts
+    protected _buildSubscriptionParams() {
+    // the `someCustomSubscription` below is the name of the subscription provided by the node your are connected to.
+        return ['someCustomSubscription', this.args];
+    }
+```
+
+With the implementation above, the call that will be made to the provider will be as follow:
+
+```ts
+{
+  id: "[GUID-STRING]", // something like: '3f839900-afdd-4553-bca7-b4e2b835c687'
+  jsonrpc: '2.0',
+  method: 'eth_subscribe',
+  // The `someCustomSubscription` below is the name of the subscription provided by the node your are connected to.
+  // And the `args` is the variable that has the type you provided at the second generic type
+  //  at your class definition. That is in the snippet above: `{customArg: string}`.
+  // And its value is what you provided when you will call:
+  //  `web3.subscriptionManager.subscribe('custom', args)`
+  params: ['someCustomSubscription', args],
+}
+```
+
+## Additional Custom Processing
+
+You may need to do some processing in the constructor. Or you may need to do some formatting for the data before it will be fired by the event emitter. In this section you can check how to do either one of those or both.
+
+### Custom Constructor
+
+You can optionally write a constructor, incase you need to do some additional initialization or processing.
+And here is an example constructor implementation:
+
+```ts
+constructor(
+  args: {customArg: string},
+  options: {
+    subscriptionManager: Web3SubscriptionManager;
+    returnFormat?: DataFormat;
+  }
+) {
+  super(args, options);
+
+  // Additional initialization
+}
+```
+
+The constructor passes the arguments to the Web3Subscription parent class.
+
+You can access the subscription manager via `this.subscriptionManager`.
+
+### Custom formatting
+
+In case you need to formate the data received from the node before it will be emitted, you just need to override the protected method `formatSubscriptionResult` in your class. It will be something like the following. However, the data type could be whatever provided by the node and it is what you should already provided at the first generic type when you extended `Web3Subscription`:
```

Comment:
```suggestion
In case you need to format the data received from the node before it will be emitted, you just need to override the protected method `formatSubscriptionResult` in your class. It will be something like the following. However, the data type could be whatever provided by the node and it is what you should already provided at the first generic type when you extended `Web3Subscription`:
```